### PR TITLE
Move Prow/Testgrid templates into individual yaml files

### PR DESCRIPTION
**Pull request by chaodaiG**
All of our Prow versions live in `*.yaml` files, other than the template inlined in `ci/prow/make_config.go`. Which makes Prow auto version bumper tool need to parse Go files. Move all templates to individual yaml files to avoid parsing Go files.

Part of: #689

### CODE REVIEW

File: ci/prow/templates/general_prow_config.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Copyright [[.Year]] The Knative Authors
```
Comment (adrcunha) [inline ci/prow/templates/general_prow_config.yaml]: I think "prow_config_header.yaml" would be better.

File: ci/prow/make_config.go [hunk 2]
```diff
@@ -218,320 +219,57 @@ var (
...
-  [[indent_array 2 .RepoNames]]
-	`
+	dashboardGroupTemplate = "testgrid_dashboardgroup.yaml"
 )
 
 // Yaml parsing helpers.
 
+// read template yaml file content
+func readTemplate(fp string) string {
+	content, err := ioutil.ReadFile(path.Join(templateDir, fp))
```
Comment (adrcunha) [inline ci/prow/make_config.go]: I can't help saying that this is I/O inefficient. :) You're reading it from disk every time you need a template. You could store them in a map once they're read, so you only read them from disk once. But I'm OK with a TODO if you don't want to do it now, this is not a critical performance issue.

File: ci/prow/templates/testgrid_general_config.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Copyright [[.Year]] The Knative Authors
```
Comment (adrcunha) [inline ci/prow/templates/testgrid_general_config.yaml]: I think "testgrid_config_header.yaml" would be better.

File: ci/prow/templates/general_prow_config.yaml (hunk 1 above)
Comment (chaodaiG) [reply ci/prow/templates/general_prow_config.yaml]: done

File: ci/prow/make_config.go (hunk 2 above)
Comment (chaodaiG) [reply ci/prow/make_config.go]: done

File: ci/prow/templates/testgrid_general_config.yaml (hunk 3 above)
Comment (chaodaiG) [reply ci/prow/templates/testgrid_general_config.yaml]: done

File: ci/prow/make_config.go [hunk 4]
```diff
@@ -213,325 +214,68 @@ var (
 	// key is the main project version, value is another map containing job details
 	//     for the job detail map, key is the repo name, value is the list of job types, like continuous, latency, nightly, and etc.
 	metaData = make(map[string]map[string][]string)
+
+	// tempaltesCache caches templates in memory to avoid I/O
```
Comment (adrcunha) [inline ci/prow/make_config.go]:
```suggestion
	// templatesCache caches templates in memory to avoid I/O
```

**Pull request merged**